docs/site/tutorials/protocol_oriented_generics.ipynb

```diff
@@ -0,0 +1,1307 @@
+        "\n",
+        "In Swift, there are different ways to express inheritence. One of those ways you may be familiar with already - class inheritence. However, Swift has another way - protocols.\n",
+        "\n",
+        "In this tutorial, we will explore protocols - an alternative to subclassing that allows you to achieve similar goals through different tradeoffs. In Swift, protocols contain multiple abstract members. Classes, structs and enums can conform to multiple protocols and the conformance relationship can be established retroactively. All that enables some designs that aren't easily expressible in Swift using subclassing. We will walk through the idioms that support the use of protocols (extensions and protocol constraints), as well as the limitations of protocols.\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "5AIIH5Q59b41",
+        "colab_type": "text"
+      },
+      "source": [
+        "## Swift 💖's value types!\n",
+        "\n",
+        "In addition to classes which have reference semantics, Swift supports enums and structs that are passed by value. Enums and structs support many features provided by classes. Let's take a look!\n",
+        "\n",
+        "Firstly let's look at how enums are similar to classes."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "zCN0Uc0w-gng",
+        "colab_type": "code",
+        "outputId": "4dbad5e2-fad0-4be9-f6f2-bf9982c94893",
+        "colab": {
+          "base_uri": "https://localhost:8080/",
+          "height": 51
+        }
+      },
+      "source": [
+        "enum Color: String {\n",
+        "    case red = \"red\"\n",
+        "    case green = \"green\"\n",
+        "    case blue = \"blue\"\n",
+        "    // A computed property. Note that enums cannot contain stored properties.\n",
+        "    var giveHint: String {\n",
+        "        switch self {\n",
+        "            case .red:\n",
+        "                return \"Roses are this color.\"\n",
+        "            case .green:\n",
+        "                return \"Grass is this color.\"\n",
+        "            case .blue:\n",
+        "                return \"The ocean is this color.\"\n",
+        "        }\n",
+        "    }\n",
+        "    \n",
+        "    // An initializer like for classes.\n",
+        "    init?(color: String) {\n",
+        "        switch color {\n",
+        "        case \"red\":\n",
+        "            self = .red\n",
+        "        case \"green\":\n",
+        "            self = .green\n",
+        "        case \"blue\":\n",
+        "            self = .blue\n",
+        "        default:\n",
+        "            return nil\n",
+        "        }\n",
+        "    }\n",
+        "}\n",
+        "\n",
+        "// Can extend the enum as well!\n",
+        "extension Color {\n",
+        "    // A function.\n",
+        "    func giveHintFunc() -> String {\n",
```

Comment:
```suggestion
        "    func hint() -> String {\n",
```

Reply:
This is an extension on `Color` so will rename it to `hintFunc` instead since we already have the `hint` computed property.